### PR TITLE
Free users cannot access larger models

### PR DESCRIPTION
**Pull request by philipperolet**
Following issue #1577 and PR #1662 

Free users now don't have access to large models (tested with free & upgraded versions, on model selection and agent list)

Remaining work: 
- Assistant list toggle => fix the behaviour
- Solution for the "contact us" (cf @Duncid comment about design in the issue)

If ok, pending the cool designed solution, I can do like we do for managed datasources =>
```
                  if (agent.status === "require_upgrade")
                    alert(
                      "Advanced assistants are only available on our paid plans. Contact us at team@dust.tt to get access."
                    );
```

### CODE REVIEW

File: front/types/assistant/agent.ts [hunk 1]
```diff
@@ -59,7 +59,8 @@ export type AgentGenerationConfigurationType = {
 export type GlobalAgentStatus =
   | "active"
   | "disabled_by_admin"
-  | "disabled_missing_datasource";
+  | "disabled_missing_datasource"
+  | "require_upgrade";
```
Comment (PopDaph) [inline front/types/assistant/agent.ts:63]: NIT: `disabled_free_workspace` to follow the convention disable + reason

**Comment by spolu** [comment]
Alert as a first intention is fine. "Advanced assistants" is not clear enought. Where is that alert going to be?

File: front/types/assistant/agent.ts (hunk 1 above)
Comment (spolu) [reply front/types/assistant/agent.ts:63]: yes let's update :+1:

**Review by spolu — DISMISSED** [review]
LGTM modulo rename of the status

no need to r? once fixed

**Comment by philipperolet** [comment]
Ok; Alert when you click on the toggle (via onclick handler, a bit ugly but since temp). 

Hesistated between Advanced Assistants & Assistants with larger models, thought the 2nd one more precise but the 1st one more user friendly

"Assistants with larger models" are not.... => ok for you?

**Comment by spolu** [comment]
"GPT4 and Claude 2 are only available on our paid plans. Contact us at team@dust.tt to get access."

**Comment by philipperolet** [comment]
> "GPT4 and Claude 2 are only available on our paid plans. Contact us at [team@dust.tt](mailto:team@dust.tt) to get access."

But dust / slack / notion etc are also only available on our paid plans => I will just put the name of the agent in the alert

File: front/pages/w/[wId]/builder/assistants/index.tsx [hunk 2]
```diff
@@ -79,7 +79,7 @@ export default function AssistantsBuilder({
   const handleToggleAgentStatus = async (agent: AgentConfigurationType) => {
     if (agent.status === "disabled_free_workspace") {
       window.alert(
-        `${agent.name} is only available on our paid plans. Contact us at team@dust.tt to get access.`
+        `Assistant @${agent.name} is only available on our paid plans. Contact us at team@dust.tt to get access.`
```
Comment (spolu) [inline front/pages/w/[wId]/builder/assistants/index.tsx:82]: Drop the Assistant :+1:

Comment (philipperolet) [reply front/pages/w/[wId]/builder/assistants/index.tsx:82]: > Drop the Assistant 👍

it's cleaner :)

**Pull request merged**